### PR TITLE
Fix javadocs

### PR DESCRIPTION
**Pull request by dalinaum**
Please review @realm/java

### CODE REVIEW

**Comment by nhachicha** [comment]
:+1:

**Comment by zaki50** [comment]
Some other comments start with "Returns".

We should not add "s"?

**Comment by dalinaum** [comment]
@zaki50 I think `Returns` should be fixed too. I will update this PR.

**Comment by emanuelez** [comment]
I think  we should keep the s everywhere.

**Comment by cmelchior** [comment]
Yes, we agreeded on that style for our Javadoc at some point. Keep the `s` for now. We need to take a sweep of all our javadoc at some point anyway because it is not entirely consistent right now. So we should take the discussion then. I added a point to this issue about it: #1594

**Comment by dalinaum** [comment]
Because I saw `Check` and `Test` many times, I replaced `Tests` with `Test`. If we should keep the s everywhere, I will modify all comments.

**Comment by cmelchior** [comment]
I am not sure it is worth doing that until we agreed on a template for the Javadoc. Right now it is a bit unspoken. It would probably be better to draft up the rules in the wiki before starting changing a lot of javadoc.

**Comment by dalinaum** [comment]
I just read @cmelchior 's comment and I already pushed a new commit. I should remove `review` until drafting up the rules in the wiki.

**Comment by dalinaum** [comment]
Modified javadoc as a javadoc guideline. Please review this again @realm/java

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 1]
```diff
@@ -402,12 +394,11 @@ private static void initializeRealm(Realm realm) {
 
     /**
      * Create a Realm object for each object in a JSON array. This must be done within a transaction.
```
Comment (zaki50) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: `Creates`?

Comment (dalinaum) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Thanks. I should fix it.

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 2]
```diff
@@ -164,19 +162,17 @@ protected void finalize() throws Throwable {
...
      *
      * @param context a non-null Android {@link android.content.Context}
      * @return an instance of the Realm class.
-
      * @throws java.lang.IllegalArgumentException if no {@link Context} is provided.
      * @throws RealmMigrationNeededException if the model classes no longer match the underlying Realm
-     *                                       and it must be migrated.
-     * @throws RealmIOException              if an error happened when accessing the underlying Realm
-     *                                       file.
+     *      and it must be migrated.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Spacing looks a bit weird here?

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 3]
```diff
@@ -189,10 +185,9 @@ public static Realm getInstance(Context context) {
      * by {@link #setDefaultConfiguration(RealmConfiguration)}
      *
      * @return an instance of the Realm class.
-     *
-     * @throws java.lang.NullPointerException If no default configuration has been defined.
-     * @throws RealmMigrationNeededException If no migration has been provided by the default configuration and the
-     * model classes or version has has changed so a migration is required.
+     * @throws java.lang.NullPointerException if no default configuration has been defined.
+     * @throws RealmMigrationNeededException if no migration has been provided by the default configuration and the
+     *      model classes or version has has changed so a migration is required.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Same

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 4]
```diff
@@ -206,9 +201,8 @@ public static Realm getDefaultInstance() {
      *
      * @param configuration {@link RealmConfiguration} used to open the Realm
      * @return an instance of the Realm class
-     *
-     * @throws RealmMigrationNeededException If no migration has been provided by the configuration and the
-     * model classes or version has has changed so a migration is required.
+     * @throws RealmMigrationNeededException if no migration has been provided by the configuration and the model
+     *      classes or version has has changed so a migration is required.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Same

Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Basically all the places. Do we indent second line like this?

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 5]
```diff
@@ -832,12 +810,11 @@ boolean contains(Class<? extends RealmObject> clazz) {
     }
 
     /**
-     * Get all objects of a specific Class. If no objects exist, the returned RealmResults will not
-     * be null. The RealmResults.size() to check the number of objects instead.
+     * Gets all objects of a specific Class. If no objects exist, the returned RealmResults will not be null. The
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: {@code null}.

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 6]
```diff
@@ -845,13 +822,13 @@ boolean contains(Class<? extends RealmObject> clazz) {
     }
 
     /**
-     * Get all objects of a specific Class sorted by a field.  If no objects exist, the returned
-     * RealmResults will not be null. The RealmResults.size() to check the number of objects instead.
+     * Get all objects of a specific Class sorted by a field. If no objects exist, the returned {@link RealmResults}
+     * will not be null. The RealmResults.size() to check the number of objects instead.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: {@code null}.

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 7]
```diff
@@ -870,16 +847,15 @@ boolean contains(Class<? extends RealmObject> clazz) {
 
 
     /**
-     * Get all objects of a specific class sorted by two specific field names.  If no objects exist,
-     * the returned RealmResults will not be null. The RealmResults.size() to check the number of
-     * objects instead.
+     * Gets all objects of a specific class sorted by two specific field names.  If no objects exist, the returned
+     * {@link RealmResults} will not be null. The RealmResults.size() to check the number of objects instead.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: {@code null}.

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 8]
```diff
@@ -1169,11 +1142,11 @@ public static void migrateRealm(RealmConfiguration configuration) {
     }
 
     /**
-     * Manually trigger a migration on a RealmMigration.
+     * Manually triggers a migration on a RealmMigration.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: we should not add s to triggers here.

File: realm/realm-library/src/main/java/io/realm/RealmResults.java [hunk 9]
```diff
@@ -240,10 +239,10 @@ private long getColumnIndex(String fieldName) {
     }
 
     /**
-     * Sort (ascending) an existing {@link io.realm.RealmResults}.
+     * Sorts (ascending) an existing {@link io.realm.RealmResults}.
      *
-     * @param fieldName  The field name to sort by. Only fields of type boolean, short, int, long,
-     *                   float, double, Date, and String are supported.
+     * @param fieldName the field name to sort by. Only fields of type boolean, short, int, long, float, double, Date,
+     *                  and String are supported.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmResults.java]: I missed this in the guidelines, but I'm fine with it.

File: realm/realm-library/src/main/java/io/realm/HandlerController.java [hunk 10]
```diff
@@ -322,8 +322,11 @@ public boolean handleMessage(Message message) {
     }
 
     /**
-     * This will prevent advanceReading from accidentally advancing the thread and potentially re-run the queries in this thread.
-     * @return {@code true} if there is at least one (non GC'd) instance of {@link RealmResults} {@code false} otherwise
+     * This will prevent advanceReading from accidentally advancing the thread and potentially re-run the queries in
+     * this thread.
+     *
+     * @return {@code true} if there is at least one (non GC'd) instance of {@link RealmResults} {@code false}
+     *      otherwise.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/HandlerController.java]: Indentation?

**Comment by cmelchior** [comment]
My mind sort of imploded going through this, so a small :+1: 
In any case, even if this introduces a few errors, the number of errors fixed will be far far greater.

**Comment by dalinaum** [comment]
@cmelchior Android Studio aligns descriptions of parameters automatically, while it dont' align others. I just added two tabs on them. Should I align them as a first character of descriptions?

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java [hunk 11]
```diff
@@ -282,15 +286,15 @@ public void refresh() {
     }
 
     /**
-     * Starts a transaction, this must be closed with {@link io.realm.Realm#commitTransaction()} or
-     * aborted by {@link io.realm.Realm#cancelTransaction()}. Transactions are used to atomically
-     * create, update and delete objects within a realm.
+     * Starts a transaction, this must be closed with {@link io.realm.Realm#commitTransaction()} or aborted by
+     * {@link io.realm.Realm#cancelTransaction()}. Transactions are used to atomically create, update and delete objects
+     * within a realm.
```
Comment (kneth) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: "realm" -> "Realm"

Comment (dalinaum) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: Thanks!

**Comment by dalinaum** [comment]
Please review this again @realm/java

**Comment by cmelchior** [comment]
:+1:

**Pull request merged**